### ISSUE TITLE
Attributes loop is extremely slow

### ISSUE THREAD

**Issue by kwkroll32** [blachlylab/gff3#2]
cutting the attributes loop from the reader reduces runtime from 62 seconds to 9 seconds

in reader.go

<pre>
...
rec.AttributesField = make(map[string]string)
    var kv []string
    for _, attribute := range strings.Split(fields[8], ";") {
        kv = strings.Split(attribute, "=")
        rec.AttributesField[kv[0]] = kv[1]
    }
</pre>

**Comment by kwkroll32** [comment]
This has been greatly improved using strings.Index instead of Split, but still takes 30 seconds. 

It could still be improved by only parsing 0-3 important fields with Index and storing the rest of the string as an attribute of the struct [i.e. do populate gff3.Record.AttributesField["gene_id"], but store the rest of the field like, `gff3.Record.AttributeString = 'ID=ENSG00001.1;gene_name=ASDF;...'`]. Then, implement a parseAttributes method for gff3.Record that fills in the rest of the remaining attributes into the struct. The idea is that most users won't need access to ALL values in the gff3 attribute fields, but most users parsing gff3 could use gene name etc. Don't populate the Record object with information they don't need, but give gff3.Record the functionality to retrieve it after the line is parsed.

**Comment by jblachly** [comment]
I agree and will implement a lazy evaluation so that it is parsed when requested
